adapters/repos/db/inverted/searcher.go

```diff
@@ -170,7 +169,16 @@ func (s *Searcher) objectsByDocID(it docIDsIterator,
 func (s *Searcher) DocIDs(ctx context.Context, filter *filters.LocalFilter,
 	additional additional.Properties, className schema.ClassName,
 ) (helpers.AllowList, error) {
-	return s.docIDs(ctx, filter, additional, className, 0)
+	allow, err := s.docIDs(ctx, filter, additional, className, 0)
+	if err != nil {
+		return nil, err
+	}
+	// Some filters, such as NotEqual, return a theoretical range of docIDs
+	// which also includes a buffer in the underlying bitmap, to reduce the
+	// overhead of repopulating the base bitmap. Here we can truncate that
+	// buffer to ensure that the caller is receiving only the possible range
+	// of docIDs
+	return allow.Truncate(s.bitmapFactory.ActualMaxVal()), nil
```

Comment:
maybe truncate should be performed by bitmap factory on cloning step (to hide internals from the user and make ux little more friendly). Or have buffer removed entirely as truncating may be more expensive than populating on each request (which we wanted to avoid with buffer).